csharp/src/Apache.Arrow.Flight/Internal/FlightRecordBatchStreamReader.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Apache.Arrow.Flatbuf;
+using Apache.Arrow.Flight.Protocol;
+using Apache.Arrow.Ipc;
+using Google.Protobuf;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight.Internal
+{
+    /// <summary>
+    /// Stream of record batches
+    ///
+    /// Use MoveNext() and Current to iterate over the batches.
+    /// There are also gRPC helper functions such as ToListAsync() etc.
+    /// </summary>
+    public abstract class FlightRecordBatchStreamReader : IAsyncStreamReader<RecordBatch>
```

Comment:
Can this class also implement `IAsyncEnumerable<RecordBatch>`? Then callers can just directly `await foreach` it, without having to call `ReadAllAsync()`.

My understanding of why `IAsyncStreamReader<T>` doesn't implement `IAsyncEnumerable<T>` is because the Grpc interface came first, and it couldn't be changed without a breaking change. @JamesNK - is this the case?

Reply:
Implemented IAsyncEnumerable